src/clp_ffi_py/ir/native/decoding_methods.cpp

```diff
@@ -42,6 +47,35 @@ concept TerminateHandlerSignature = requires(TerminateHandler handler) {
     } -> std::same_as<bool>;
 };
 
+/**
+ * Handles the error when IRErrorCode::IRErrorCode_Incomplete_IR is seen. The handler will first
+ * try to load more data into `decoder_buffer`. If it fails, `allow_incomplete_stream` will be used
+ * to determine whether to swallow the incomplete IR exception.
+ * @param decoder_buffer
+ * @param allow_incomplete_stream A flag to indicate whether the incomplete stream error should be
+ * ignored. If it is set to true, incomplete stream error
+ * @param std::nullopt if more data is loaded.
+ * @param PyNone if the IR stream is incomplete and allowed.
+ * @param nullptr if the IR stream is incomplete not allowed, with the relevant Python exceptions
+ * and error set.
+ */
```

Comment:
This part of comments adapts to our latest guideline, where the comments are also 100 char length. I will send a PR after this one to update the rest of comments.